lib/util/timeout.js

```diff
@@ -7,11 +7,21 @@ class KnexTimeoutError extends Error {
   }
 }
 
+function timeout(promise, ms) {
+  return new Promise(async function(resolve, reject) {
```

Comment:
I guess I need to rewatch https://www.youtube.com/watch?v=e3Nh350b6S4&feature=youtu.be, but wasn't using async inside promise callback kinda dangerous?

Reply:
Yeah -- it can be dangerous if there is a bug in your error handling.  So, I think that's the reason they advise against it.

In either case: I already rewrote the logic to do the same thing without `async`.